stress/basic.go

```diff
@@ -0,0 +1,539 @@
+package stress
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdb/influxdb/client/v2"
+)
+
+// AbstractTag is a struct that contains data
+// about a tag for in a series
+type AbstractTag struct {
+	Key   string `toml:"key"`
+	Value string `toml:"value"`
+}
+
+// AbstractTags is an slice of abstract tags
+type AbstractTags []AbstractTag
+
+func (a AbstractTags) Tag(i int) Tags {
+	tags := make(Tags, len(a))
+
+	for j, t := range a {
+		tags[j] = Tag{
+			Key:   t.Key,
+			Value: fmt.Sprintf("%v-%v", t.Value, i),
+		}
+	}
+
+	return tags
+}
+
+// Template returns a templated string of tags
+func (t AbstractTags) Template() string {
+	var buf bytes.Buffer
+	for i, tag := range t {
+		if i == 0 {
+			buf.Write([]byte(fmt.Sprintf("%v=%v-%%v,", tag.Key, tag.Value)))
+		} else {
+			buf.Write([]byte(fmt.Sprintf("%v=%v,", tag.Key, tag.Value)))
+		}
+	}
+
+	b := buf.Bytes()
+	b = b[0 : len(b)-1]
+
+	return string(b)
+}
+
+// tag is a struct that contains data
+// about a field for in a series
+type AbstractField struct {
+	Key  string `toml:"key"`
+	Type string `toml:"type"`
+}
+
+// AbstractFields is an slice of abstract fields
+type AbstractFields []AbstractField
+
+func (a AbstractFields) Field() Fields {
+	fields := make(Fields, len(a))
+
+	for i, f := range a {
+		field := Field{Key: f.Key}
+		switch f.Type {
+		case "float64":
+			field.Value = fmt.Sprintf("%v", rand.Intn(1000))
+		case "int":
+			field.Value = fmt.Sprintf("%vi", rand.Intn(1000))
+		case "bool":
+			b := rand.Intn(2) == 1
+			field.Value = fmt.Sprintf("%v", b)
+		default:
+			field.Value = fmt.Sprintf("%v", rand.Intn(1000))
+		}
+
+		fields[i] = field
+	}
+
+	return fields
+}
+
+// Template returns a templated string of fields
+func (f AbstractFields) Template() (string, []string) {
+	var buf bytes.Buffer
+	a := make([]string, len(f))
+	for i, field := range f {
+		buf.Write([]byte(fmt.Sprintf("%v=%%v,", field.Key)))
+		a[i] = field.Type
+	}
+
+	b := buf.Bytes()
+	b = b[0 : len(b)-1]
```

Comment:
Give the two `AbstractFields`

```
{key: "value", type:"float64"}
```
```
{key: "other_value", type: "float64"}
```

Template will return `value=%v,other_value=%v`.

Lines 100 and 101 trim the trailing comma off of `value=%v,other_value=%v,`